File: HowToFixDuplicateCells/HowToFixDuplicateCells/ViewController.m
//
//  ViewController.m
//  HowToFixDuplicateCells
//
//  Created by Ibrahim Hassan on 17/01/19.
//  Copyright © 2019 None. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    NSMutableArray *tempArr = [[NSMutableArray alloc] init];
    for (int i = 0; i < 100; i++) {
        [tempArr addObject: [[NSString alloc] initWithFormat:@"Cell %d", i]];
    }
    tableData = tempArr;
}


- (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"MyCell" forIndexPath:indexPath];
    if (indexPath.row % 2 == 1) {
        cell.backgroundColor = UIColor.grayColor;
    }
    /*
     else {
     cell.backgroundColor = UIColor.whiteColor;
     }
     */
    cell.textLabel.text = tableData[indexPath.row];
    return cell;
}

- (NSInteger)tableView:(nonnull UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return tableData.count;
}


@end

File: HowToFixDuplicateCells/HowToFixDuplicateCells-Swift/ViewController.swift
//
//  ViewController.swift
//  HowToFixDuplicateCells-Swift
//
//  Created by Ibrahim Hassan on 17/01/19.
//  Copyright © 2019 None. All rights reserved.
//

import UIKit

class SwiftViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
    var tableData: [Any] = []
    
    override func viewDidLoad() {
        super.viewDidLoad()
        var tempArr: [AnyHashable] = []
        for i in 0..<100 {
            tempArr.append("Cell \(i)")
        }
        tableData = tempArr
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell: UITableViewCell = tableView.dequeueReusableCell(withIdentifier: "MyCell", for: indexPath)
        if indexPath.row % 2 == 0 {
            cell.backgroundColor = UIColor.gray
        }
        /*
         else {
         cell.backgroundColor = UIColor.white
         }
         */
        cell.textLabel?.text = tableData[indexPath.row] as? String
        return cell
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return tableData.count
    }
}

